app/my-bets/page.tsx

```diff
@@ -7,96 +7,119 @@ import { Market, Position, tradesOutcomeBalance } from '@/entities';
 
 import { getUserPositions } from '@/queries/conditional-tokens';
 import { UserPosition } from '@/queries/conditional-tokens/types';
-import { getConditionMarket, getMarketUserTrades } from '@/queries/omen';
-import { useQuery, useQueryClient } from '@tanstack/react-query';
-
-import { PropsWithChildren, ReactNode, useCallback, useEffect, useState } from 'react';
+import {
+  Condition,
+  FixedProductMarketMaker,
+  FpmmTrade,
+  getConditionMarket,
+  getMarketUserTrades,
+} from '@/queries/omen';
+import { useQuery } from '@tanstack/react-query';
+
+import { PropsWithChildren, ReactNode, useMemo } from 'react';
 import { TabBody, TabGroup, TabHeader, TabPanel, TabStyled } from '@swapr/ui';
 import { useAccount } from 'wagmi';
 
+export interface UserPositionComplete extends UserPosition {
+  fpmmTrades: FpmmTrade[];
+  market: FixedProductMarketMaker;
+  condition: Condition;
+}
+
 export default function MyBetsPage() {
   const { address } = useAccount();
-  const [unredeemedBets, setUnredeemedBets] = useState<UserPosition[]>([]);
-  const queryClient = useQueryClient();
 
-  const { data, isLoading } = useQuery({
-    queryKey: ['getUserPositions', address],
-    queryFn: () => getUserPositions({ id: address?.toLowerCase() as string }),
-    enabled: !!address,
-  });
+  const { data: userPositionsComplete, isLoading } = useQuery({
+    queryKey: ['getUserPositionsComplete', address],
+    queryFn: async () => {
+      if (!address) return [];
 
-  const userPositions = data?.userPositions ?? [];
+      const userPositionsData = await getUserPositions({ id: address.toLowerCase() });
+      const userPositions = userPositionsData?.userPositions ?? [];
 
-  const filterActiveBets = userPositions.filter(
-    position => position.position.conditions[0].resolved === false
-  );
-  const filterCompleteBets = userPositions.filter(
-    position => position.position.conditions[0].resolved
-  );
+      const userPositionsComplete = await Promise.all(
```

Comment:
Should we use `Promise.all` or `Promise.allSettled` ?

Reply:
This is a good Q.

So to summarize, IMHO we should use `Promise.allSettled` because if some of the promises fail, the execution will get stopped with our current PR approach. But using `.allSettled` we could handle each response individually, so with this update, we can show the bets that didn't fail, and handle the ones that failed.

Reply:
Changed to `allSettled`, although is a bit more complex (cause you have to deal with unfulfilled values), the benefit of showing something instead of nothing if some fail seems worth it.

Updated it.